### DISCUSSION TITLE
Documentation (swagger) not updated

### DISCUSSION THREAD

**Discussion by PaulRichez** [ComfortablyCoding/strapi-plugin-transformer#30] [Ideas]
Work, well play, i search for it then i found this project on the strapi market place :)

But ...
![image](https://user-images.githubusercontent.com/37926689/173196553-3c091056-17d1-4073-9d47-3600aec70d57.png)

On my documentation (swagger) , i can see Data and Attributes object, possible to change swagger too ?

Thanks

**Comment by ComfortablyCoding** [comment]
I will need to check how swagger generates the sample response and see if it's feasible. Most likely it's something that needs to be accounted for on swagger plugins side.

**Comment by willempie96** [comment]
Is there an update on this?

It would be great to have this because i was planning on using the generated documentation to generate typescript types for my frontend.

**Reply by PaulRichez** [reply]
Hello @willempie96 can you explain how you will generate for you're front end ? just copy and paste or something else ?

**Reply by willempie96** [reply]
I was looking to use a npm plugin in my frontend project to generate typescript types based on the openApi documentation generated by the strapi documentation plugin. (openapi-typescript-codegen) That way i can run a single command in my frontend project after which i would have the most recent types. 
I have this working in a different project but because the openapi documentation is generated based on the default strapi format i can't use the transformer because than the types are different from the api responses i receive.

**Reply by BabyDino** [reply]
I have the same issue with typescript generated files. I use the built-in Strapi generator: https://docs.strapi.io/developer-docs/latest/development/typescript.html#generate-typings-for-project-schemas

Would it be possible to use the transformer here as well @ComfortablyCoding?

**Reply by ComfortablyCoding** [reply]
@BabyDino 

The strapi type generator command does not look to be dynamic and is instead based on the content types themselves. This means any changes made to the response (or request) via middleware, controller etc do not effect it.

As they do not provide a way (at least from what I could find) to hook into this command it is not possible to modify it either.

At the moment the best way (suggestions welcome) seems to be to create our own custom command to generate the types based on activated settings of the plugin. At this time their is no ETA for it.

**Comment by sergipt-ohpen** [comment]
I have the same problem, unfortunately I will have to uninstall the transformer plugin if it is not able to modify the documentation.
